### PR TITLE
Fix ambiguous operator calls during restore

### PR DESCRIPTION
**Pull request by rishabhtanwar29**
### Description
* The JSONB operators like `->` and `->>` become ambiguous under
`dump_restore` GUC so fixed this by adding proper casts in `sys.syslanguages` view.

Task: BABEL-3826
Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql [hunk 1]
```diff
@@ -38,6 +38,23 @@ LANGUAGE plpgsql;
  * final behaviour.
  */
 
+CREATE OR REPLACE VIEW sys.syslanguages
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql:41]: As you know, this should be moved to 3.1.0--3.2.0 when merging into the main dev branch.

File: contrib/babelfishpg_tsql/src/hooks.c [hunk 2]
```diff
@@ -2334,7 +2334,11 @@ pltsql_store_func_default_positions(ObjectAddress address, List *parameters, con
 	proctup = SearchSysCache1(PROCOID, ObjectIdGetDatum(address.objectId));
 	/* Disallow extended catalog lookup during restore */
 	if (!HeapTupleIsValid(proctup) || babelfish_dump_restore)
+	{
+		if (HeapTupleIsValid(proctup))
+			ReleaseSysCache(proctup);
 		return;
+	}
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/hooks.c:2341]: minor suggestion
```
if (babelfish_dump_restore)
    return;
proctup = SearchSysCache1(PROCOID, ObjectIdGetDatum(address.objectId));
if (!HeapTupleIsValid(proctup))
    return;
```

**Review by 2jungkook — APPROVED** [review]

File: contrib/babelfishpg_tsql/sql/sys_views.sql [hunk 3]
```diff
@@ -2186,14 +2186,14 @@ CREATE OR REPLACE VIEW sys.syslanguages
 AS
 SELECT
     lang_id AS langid,
-    CAST(lower(lang_data_jsonb ->> 'date_format') AS SYS.NCHAR(3)) AS dateformat,
-    CAST(lang_data_jsonb -> 'date_first' AS SYS.TINYINT) AS datefirst,
+    CAST(lower(lang_data_jsonb ->> 'date_format'::TEXT) AS SYS.NCHAR(3)) AS dateformat,
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/sql/sys_views.sql:2189]: these issues should impact pg_upgrade as well? if yes, why didn't we target them for Ocelot?

Comment (rishabhtanwar29) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:2189]: This issue is not there with pg_upgrade does not execute extension's SQL script.

File: contrib/babelfishpg_tsql/src/hooks.c (hunk 2 above)
Comment (rishabhtanwar29) [reply contrib/babelfishpg_tsql/src/hooks.c]: Updated in new rev.

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql (hunk 1 above)
Comment (rishabhtanwar29) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql:41]: That's right.

File: contrib/babelfishpg_tsql/sql/sys_views.sql (hunk 3 above)
Comment (surendravishnoi) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:2208]: same for is_user function?

Comment (rishabhtanwar29) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:2208]: I think accessing an invalid heap tuple is problematic and should be fixed in the latest available release but considering that ocelot release is already wrapping up and we have not seen this crash with any of the pg_upgrade tests, it should be fine to put the fix in following release.

Comment (surendravishnoi) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:2208]: I think we have enough time for bug fixes, this one looks safe to put in there

Comment (rishabhtanwar29) [reply contrib/babelfishpg_tsql/sql/sys_views.sql:2212]: As discussed offline, moved the fix for `is_user` crash to Ocelot.

**Review by surendravishnoi — APPROVED** [review]

**Pull request merged**